### DISCUSSION TITLE
Toggle Text Wrapping

### DISCUSSION THREAD

**Discussion by NotGeri** [wezterm/wezterm#5539] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Hey all 👋
Is there a way to make a hotkey for the text wrapping?

As an example, here are some long lines on my normal terminal and font size, which wrap to new lines:
![image](https://github.com/wez/wezterm/assets/46295973/08ffd9a1-0d15-4524-9df1-0b7283db6251)

I checked the documentation, and the only `wrap` or similar seems to be whether to wrap around tabs and such, not the text.
I think it may be possible because if I zoom out with a hotkey with `act.DecreaseFontSize, ' it no longer wraps, so it's seemingly not a "hard" wrap:
![image](https://github.com/wez/wezterm/assets/46295973/e2fae221-8c9f-4a0a-b789-621ee6f4f843)

In some cases, it'd be super helpful to keep the same font size and not wrap but allow vertical scrolling or just cut it off.

Thanks!

**Comment by wez** [comment, marked as answer]
text wrapping is processed at the time that a line is output. If a line is wrapped, then when the window is subsequently resized, the wrapping will be preserved and the text will be rewrapped to fit the new dimensions.
However, to prevent excessive resource utilization in both wrapping and logical-line-selection, there is a built-in limit on the maximum logical line length of 1024 columns.

Lines that are longer than that size will be hard broken into physical lines that will not rewrap.

**Comment by NotGeri** [comment]
Ohh I see; thank you!
